psycop/common/model_training_v2/trainer/preprocessing/steps/filters.py

```diff
@@ -16,3 +19,58 @@ def __init__(self, min_age: int, max_age: int, age_col_name: str):
 
     def apply(self, input_df: PolarsFrame_T0) -> PolarsFrame_T0:
         return input_df.filter((self.age >= self.min_age) & (self.age <= self.max_age))
+
+
+@BaselineRegistry.preprocessing.register("lookbehind_combination_filter")
+class LookbehindCombinationFilter(PresplitStep):
+    def __init__(
+        self,
+        lookbehind_combination: set[int],
+        pred_col_prefix: str,
+        logger: BaselineLogger,
+    ):
+        self.lookbehind_combination = lookbehind_combination
+        self.pred_col_prefix = pred_col_prefix
+        self.logger = logger
+
+    def apply(self, input_df: PolarsFrame_T0) -> PolarsFrame_T0:
+        pred_cols_with_lookbehind = [
+            col
+            for col in input_df.columns
+            if col.startswith("pred_") and "within" in col
+        ]
+
+        lookbehinds_in_dataset = {
+            int(re.findall(pattern=r"within_(\d+)_days", string=col)[0])
```

Comment:
We probably want to move the pattern into the __init__ as well. Makes the filter easier to understand within looking into it :-)